### ISSUE TITLE
Reopen #94 and don't use C: drive at all in installation.

### ISSUE THREAD

**Issue by DAMACANER** [docker/roadmap#464]
**Tell us about your request**
https://github.com/docker/roadmap/issues/94, this is closed due to:

> Closing this due to the new installer flag used to change the installation location. For changing where Docker Desktop stores it's data please follow progress on the following roadmap item https://github.com/docker/roadmap/issues/239

from @andrei-z-albu 

the thing is, #239 is COMPLETELY unrelevant to the #94. I and other people want to install **DOCKER DESKTOP** to another drive not **DOCKER IMAGES/CONTAINERS/VOLUMES**

```powershell
.\Docker.exe install --installation-dir=D:\Docker
```
still doesnt work. Wonder why? Because Docker Desktop STILL uses C: drive while installing. Like, jesus christ, would I install to D:, if I had enough space at C: already?

here is a runtime of

```powershell
.\Docker.exe install --installation-dir=D:\Docker
```

```powershell
Version: 4.17.0 (99724)
Sha1:
Started on: 2023/03/17 16:53:28.438
Resources: D:\resources
OS: Windows 10 Pro
Edition: Professional
Id: 2009
Build: 19045
BuildLabName: 19041.1.amd64fre.vb_release.191206-1406
File: C:\ProgramData\DockerDesktop\install-log-admin.txt
CommandLine: "D:\Docker.exe" install --installation-dir=D:\Docker
You can send feedback, including this log file, at https://github.com/docker/for-win/issues
[2023-03-17T16:53:28.635419400Z][ManifestAndExistingInstallationLoader][I] Install path is D:\Docker. Loading manifest first
[2023-03-17T16:53:28.644420200Z][ManifestAndExistingInstallationLoader][I] No manifest found, returning no existing install
[2023-03-17T16:53:28.644420200Z][Installer][I] No installation found
[2023-03-17T16:53:28.872417900Z][InstallWorkflow][I] Cancel pending background download
[2023-03-17T16:53:28.875418500Z][BackgroundTransfer][I] Cancel current background transfer job
[2023-03-17T16:53:28.886421200Z][InstallWorkflow][I] Using package: res:DockerDesktop
[2023-03-17T16:53:28.887419200Z][InstallWorkflow][I] Downloading
[2023-03-17T16:53:29.074420900Z][InstallWorkflow][E] Installation failed System.Exception: Download failed: Diskte yeterli yer yok.
 ---> System.IO.IOException: Diskte yeterli yer yok.

   konum: System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   konum: System.IO.FileStream.WriteCore(Byte[] buffer, Int32 offset, Int32 count)
   konum: System.IO.Stream.<>c.<BeginWriteInternal>b__46_0(Object <p0>)
   konum: System.Threading.Tasks.Task`1.InnerInvoke()
   konum: System.Threading.Tasks.Task.Execute()
--- Özel durumun oluşturulduğu önceki konumdan başlayan yığın izlemesinin sonu ---
   konum: System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   konum: System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   konum: System.IO.Stream.EndWrite(IAsyncResult asyncResult)
   konum: System.IO.Stream.<>c.<BeginEndWriteAsync>b__53_1(Stream stream, IAsyncResult asyncResult)
   konum: System.Threading.Tasks.TaskFactory`1.FromAsyncTrimPromise`1.Complete(TInstance thisRef, Func`3 endMethod, IAsyncResult asyncResult, Boolean requiresSynchronization)
--- Özel durumun oluşturulduğu önceki konumdan başlayan yığın izlemesinin sonu ---
   konum: System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   konum: System.IO.Stream.<CopyToAsyncInternal>d__27.MoveNext()
--- Özel durumun oluşturulduğu önceki konumdan başlayan yığın izlemesinin sonu ---
   konum: System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   konum: System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   konum: CommunityInstaller.DownloadStep.<DoAsync>d__42.MoveNext()
--- Özel durumun oluşturulduğu önceki konumdan başlayan yığın izlemesinin sonu ---
   konum: System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   konum: System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   konum: CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__23.MoveNext()
   --- İç özel durum yığını izlemesinin sonu ---
   konum: CommunityInstaller.InstallWorkflow.<DoProcessAsync>d__23.MoveNext()
```

Özel durumun oluşturulduğu önceki konumdan başlayan yığın izlemesinin sonu => end of stacktrace

Download failed: Diskte yeterli yer yok. => not enough space

**Comment by canercetin-randomguy** [comment]
And you know what is funny? Docker doesn't release back the file space if the installation fails. So I am searching where are the leftover files for deleting.

**Comment by canercetin-randomguy** [comment]
```powershell
mklink /j "C:\Users\{USERNAME}\AppData\Local\Temp\DockerDesktop" "D:\Docker\tmp"
```

Points temporary Docker Desktop installation from C: to D: and eliminates C: fully.

Yes, easy solution, but this should not take me 2398590348765 different Google Searches.

**Comment by christophermclellan** [comment]
Hi @DAMACANER , thanks for raising this and apologies that the installation drive functionality is not working correctly I've reopened https://github.com/docker/roadmap/issues/94 and we will also take a look at the tangential issue of the [disk image location functionality not working](https://github.com/docker/for-win/issues/13269). I'm going to close this ticket for now, as issue 94 will continue to be the source of truth for progress on this.

**Comment by DAMACANER** [comment]
This problem is solved in #94, yeehoo.
